snuba/snuba_migrations/generic_metrics/0041_adjust_partitioning_meta_tables.py

```diff
@@ -0,0 +1,332 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import AggregateFunction, Column, DateTime, String, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.operations import OperationTarget, SqlOperation
+from snuba.utils.schemas import Float
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+    granularity = "8192"
+    meta_view_name = "generic_metric_counters_meta_mv"
+    meta_local_table_name = "generic_metric_counters_meta_local"
+    meta_dist_table_name = "generic_metric_counters_meta_dist"
```

Comment:
Since you are dropping tables and recreating them immediately (with the same names), I am not sure how zookeeper would react to it. It is quite possible that it may barf because it sees in its metadata that a table with this name already exists.

A saner way to do this would be to create the mv's and aggregated tables with new names. That way you can completely avoid this problem.

Or alternatively, perform drop tables in 1 migration. And do create tables in another migration.

Reply:
That makes sense. I was hoping I could get away with it. I updated to use new tables names.